pennylane/noise/mitigate.py

```diff
@@ -226,34 +191,28 @@ def qfunc(op):
     num_global_folds, fraction_scale = _divmod(scale_factor - 1, 2)
 
     n_ops = len(base_ops)
-    num_to_fold = int(round(fraction_scale * n_ops / 2))
+    num_to_fold = int(math.round(fraction_scale * n_ops / 2))
 
-    # Create new_circuit from folded list
-    with AnnotatedQueue() as new_circuit_q:
-        # Original U
-        for op in base_ops:
-            qfunc(op)
+    ops = []
+    ops.extend((copy.copy(op) for op in base_ops))
 
-        # Folding U => U (U^H U)**n.
-        for _ in range(int(num_global_folds)):
-            for op in base_ops[::-1]:
-                adjoint(qfunc)(op)
+    for _ in range(num_global_folds):
+        ops.extend((adjoint(op) for op in reversed(base_ops)))
+        ops.extend((copy.copy(op) for op in base_ops))
```

Comment:
```suggestion
        ops += base_ops
```
Just a bit more performant.